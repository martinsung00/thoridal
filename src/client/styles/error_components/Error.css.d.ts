export const errorState: string;
export const errorButton: string;
export const errorButtonTag: string;
