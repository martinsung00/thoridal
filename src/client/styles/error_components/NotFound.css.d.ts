export const notFoundState: string;
export const notFound: string;
export const notFoundTag: string;
