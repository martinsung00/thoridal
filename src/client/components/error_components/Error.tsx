import React from "react";
import styles from "../../styles/error_components/Error.css";

interface Props {
  prevState: Function;
}

export const ErrorPage: React.FunctionComponent<Props> = (props: Props) => {
  return (
    <div className={styles.errorState}>
      <div>Uh Oh.. Something Went Terribly Wrong!</div>
      <div className={styles.errorButton}>
        <div
          className={styles.errorButtonTag}
          id="return-button"
          onClick={function (
            event: React.MouseEvent<HTMLDivElement, MouseEvent>
          ) {
            props.prevState(event);
          }}
        >
          Go Back
        </div>
      </div>
    </div>
  );
};
