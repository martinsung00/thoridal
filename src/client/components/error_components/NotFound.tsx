import React from "react";
import styles from "../../styles/error_components/NotFound.css";

interface Props {
  prevState: Function;
}

export const NotFoundPage: React.FunctionComponent<Props> = (props: Props) => {
  return (
    <div className={styles.notFoundState}>
      <div>Trade Not Found</div>
      <div
        className={styles.notFound}
        id="try-again-button"
        onClick={function (
          event: React.MouseEvent<HTMLDivElement, MouseEvent>
        ) {
          props.prevState(event);
        }}
      >
        <div className={styles.notFoundTag}>Try Again</div>
      </div>
    </div>
  );
};
