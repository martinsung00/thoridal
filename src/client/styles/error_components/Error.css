.errorState {
  position: relative;
  font-size: 30px;
  font-family: Helvetica, Arial, sans-serif;
  color: black;
  top: 5em;
  left: 15%;
}

.errorButton {
  position: relative;
  top: 5em;
  left: 10em;
  height: 1.85em;
  width: 6em;
  background-color: black;
  color: rgb(255, 255, 255);
  font-size: 20px;
  text-align: center;
  border-radius: 3px;
  border: 1.3px solid darkgray;
  cursor: pointer;
}

.errorButton:hover {
  position: relative;
  top: 5em;
  left: 10em;
  height: 1.85em;
  width: 6em;
  background-color: rgb(255, 255, 255);
  color: rgb(0, 0, 0);
  font-size: 20px;
  text-align: center;
  border-radius: 3px;
  border: 1.3px solid black;
  cursor: pointer;
  transition: ease-out 0.15s;
}

.errorButtonTag {
  position: relative;
  margin-top: 0.4em;
}
