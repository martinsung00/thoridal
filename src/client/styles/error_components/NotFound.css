.notFoundState {
  position: relative;
  font-size: 30px;
  font-family: Helvetica, Arial, sans-serif;
  text-align: center;
  color: black;
  top: 5em;
  left: 0;
}

.notFound {
  position: relative;
  float: left;
  top: 5em;
  left: 16em;
  height: 1.85em;
  width: 6em;
  background-color: rgb(0, 0, 0);
  color: rgb(255, 255, 255);
  font-size: 20px;
  text-align: center;
  border-radius: 3px;
  border: 1.3px solid rgb(0, 0, 0);
  cursor: pointer;
}

.notFound:hover {
  position: relative;
  float: left;
  top: 5em;
  left: 16em;
  height: 1.85em;
  width: 6em;
  background-color: darkgray;
  color: rgb(0, 0, 0);
  font-size: 20px;
  text-align: center;
  border-radius: 3px;
  border: 1.3px solid black;
  cursor: pointer;
  transition: ease-out 0.15s;
}

.notFoundTag {
  position: relative;
  margin-top: 0.4em;
}
