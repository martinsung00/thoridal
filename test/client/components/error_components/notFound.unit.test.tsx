import React from "react";
import renderer from "react-test-renderer";
import { NotFoundPage } from "../../../../src/client/components/error_components/NotFound";
import { shallow, configure, ShallowWrapper } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("Error Page", function () {
  const flushPromises = () => new Promise(setImmediate);
  const props = {
    prevState: jest.fn(),
  };
  const event = {
    currentTarget: {},
  } as React.MouseEvent<HTMLInputElement>;

  afterAll(function () {
    jest.resetAllMocks();
  });

  it("should render the page as is", function (done) {
    const container:
      | renderer.ReactTestRendererJSON
      | renderer.ReactTestRendererJSON[]
      | null = renderer.create(<NotFoundPage {...props} />).toJSON();

    expect(container).toMatchSnapshot();

    done();
  });

  describe("return button", function () {
    it("should call the prevState method", async function (done) {
      const wrapper: ShallowWrapper<
        any,
        Readonly<{}>,
        React.Component<{}, {}, any>
      > = shallow(<NotFoundPage {...props} />);

      const spy = jest.spyOn<any, any>(props, "prevState");

      event.preventDefault = jest.fn();

      wrapper.find("#try-again-button").simulate("click", event);
      await flushPromises();
      wrapper.update();

      expect(spy).toHaveBeenCalled();
      expect(wrapper).toMatchSnapshot();

      wrapper.unmount();

      done();
    });
  });
});
