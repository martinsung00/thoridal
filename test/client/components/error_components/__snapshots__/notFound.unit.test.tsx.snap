// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Page return button should call the prevState method 1`] = `ShallowWrapper {}`;

exports[`Error Page should render the page as is 1`] = `
<div>
  <div>
    Trade Not Found
  </div>
  <div
    id="try-again-button"
    onClick={[Function]}
  >
    <div>
      Try Again
    </div>
  </div>
</div>
`;
